### ISSUE TITLE
How to calculate Value of returnChange...

### ISSUE THREAD

**Issue by minzak** [angelstakepool/add-hw-wallet-owner-to-pool#3]
I'm trying to rebuild one-owned mainnet pool to multi-owner pool.
Here is [link to pool](https://cardanoscan.io/pool/dc7c3a976125562ba3238d8909bd37317b6407fd3534686871286b13?tab=epochhistory)
And I do not fully understand how to calculate needed values.

Here is your part of the code with the 50th row, how to calculate `returnChange` here, if I have worked node with an `active balance` near 67ADA, and want to set/transfer 20ADA from this balance to second milti-owner? 
```
cardano-cli transaction build-raw \
     --tx-in <utxo> \
     --tx-out $(cat payment.addr)+returnChange \
     --invalid-hereafter <ttl> \
     --fee <fee> \
     --certificate-file pool.cert \
     --out-file tx-pool.raw
```

Because I'm trying to calculate various sum, and there is no luck for me.
Here error if I use just a `0`

`Command failed: transaction submit  Error: Error while submitting tx: ShelleyTxValidationError ShelleyBasedEraAlonzo (ApplyTxError [DelegsFailure (DelplFailure (DelegFailure (StakeDelegationImpossibleDELEG (KeyHashObj (KeyHash "9e96afb947f529efa5b41f03933f7b117fa8fa2535e601d02404ba15")))))])`

or here error if I use the lowest value from 67ADA:

`Command failed: transaction submit  Error: Error while submitting tx: ShelleyTxValidationError ShelleyBasedEraAlonzo (ApplyTxError [UtxowFailure (WrappedShelleyEraFailure (UtxoFailure (ValueNotConservedUTxO (Value 67621830 (fromList [])) (Value 42621830 (fromList []))))),DelegsFailure (DelplFailure (DelegFailure (StakeDelegationImpossibleDELEG (KeyHashObj (KeyHash "9e96afb947f529efa5b41f03933f7b117fa8fa2535e601d02404ba15")))))])`

My full code is here:
[expand_pool.sh.txt](https://github.com/angelstakepool/add-hw-wallet-owner-to-pool/files/8833414/expand_pool.sh.txt)

Thanks.

**Comment by angelstakepool** [comment]
I see that your script is not related to this github repo but ... anyways, your problem, from my limited perspective, seems to be a matter of utxos. 

A. if you are spending an utxo with 5,000000 lovelaces ( 5 ADA), you want to process a tx and want the change back in the same address, then the returnChange would be 5,000000 - fee
B. if you are spending an utxo with 5,000000 lovelaces, you for example want to send 2,000000 lovelaces (2 ADA) to addr1 and the rest to addr2

it would be something like:

utxo1: 5,000000 loveces
fee = 180000 lovelaces
--tx-out $(cat addr1)+2000000
--tx-out $(cat addr2)+2982000

basically values of utxos must be conserved when you are building a tx manually.  

ps: I did not analyze your bash script in details, but in --tx-in you pass the value for a given utxo , not the account balance

hope this helps, @ANGELpool (telegram)
